library/worldgen/material_rule/src/main/java/org/quiltmc/qsl/worldgen/material_rule/impl/SurfaceMaterialRuleContextImpl.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.worldgen.material_rule.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.world.gen.surfacebuilder.SurfaceRules;
+
+import org.quiltmc.qsl.worldgen.material_rule.api.SurfaceMaterialRuleContext;
+
+@ApiStatus.Internal
+public class SurfaceMaterialRuleContextImpl implements SurfaceMaterialRuleContext, SurfaceMaterialRuleContext.Nether, SurfaceMaterialRuleContext.TheEnd {
+	private final SurfaceRules.SequenceMaterialRule sequenceRule;
+
+	public SurfaceMaterialRuleContextImpl(SurfaceRules.MaterialRule rules) {
+		this.sequenceRule = new SurfaceRules.SequenceMaterialRule(new ArrayList<>());
+		this.materialRules().add(rules);
+	}
+
+	@Override
+	public List<SurfaceRules.MaterialRule> materialRules() {
+		return this.sequenceRule.sequence();
+	}
+
+	SurfaceRules.SequenceMaterialRule getSequenceRule() {
```

Comment:
Make the package visibility on methods like these explicit on this class

Reply:
This is the explicit visibility already

They are package-private and intended to be package-private.

Reply:
This is implicit, and yeah, while I understand the use of them for things like nested classes and interfaces, it doesn't really make sense for methods and yeah, it just makes the code confusing and inconsistent

Reply:
After discussing on Discord, I came to the conclusion that this is fine; Marking this as resolved